src/dcp/dcp.c

```diff
@@ -271,10 +425,97 @@ int main(int argc, char** argv)
         usage = 1;
     }
 
+    char** argpaths = (&argv[optind]);
+
+#ifdef DAOS_SUPPORT
+    rc = daos_init();
```

Comment:
To be extra safe, let's add an ```if (mfu_src_file->type == DAOS || mfu_dst_file->type == DAOS)``` check around the code in this section so that it's not executed unless someone specified DAOS.

Reply:
@adammoody well that will mess up the case where the unified namespaces are used, because I cannot fill in the types until after daos is initialized and then running some daos code on it to check if it is a "daos path"

Reply:
@adammoody that would be ok if the src and dest pools/cont uuids were always passed in by cmd line, but that isn't always the case. You can also create POSIX DAOS containers that are linked to regular POSIX paths, and in that case you have to run DAOS code to lookup the uuids to check if it is actually within DAOS.

Reply:
Ah, I see.  That makes sense.

Reply:
We might be able to use something like stat or statfs in the future to help.  As part of the v0.11 release, I plan to look at a way to identify the underlying file system for each path.  This check will likely go in the mfu_param_path_set_all function.  We need that for things like finding the Lustre version or to get properties like the alignments needed for O_DIRECT.  We might be able to conditionally daos_init at that point, or we might need to pull the daos_init over to that part.

We can revisit that idea later.